smoketest/scripts/cli/test_interfaces_wireless.py

```diff
@@ -146,30 +146,117 @@ def test_wireless_hostapd_config(self):
+        tmp = get_config_value(interface, 'ieee80211h')
+        self.assertEqual('1', tmp)
+        tmp = get_config_value(interface, 'ieee80211ax')
+        self.assertEqual('1', tmp)
+
+        # channel and channel width
+        tmp = get_config_value(interface, 'channel')
+        self.assertEqual(channel, tmp)
+        tmp = get_config_value(interface, 'op_class')
+        self.assertEqual(channel_set_width, tmp)
+        tmp = get_config_value(interface, 'he_oper_centr_freq_seg0_idx')
+        self.assertEqual(center_channel_freq_1, tmp)
+
+        # Country code
+        tmp = get_config_value(interface, 'country_code')
+        self.assertEqual(country.upper(), tmp)
+
+        # BSS coloring
+        tmp = get_config_value(interface, 'he_bss_color')
+        self.assertEqual(bss_color, tmp)
+
+        # sae_password
+        tmp = get_config_value(interface, 'sae_password')
+        self.assertEqual(sae_pw, tmp)
+
+        # WPA3 and dependencies
+        tmp = get_config_value(interface, 'wpa')
+        self.assertEqual('2', tmp)
+        tmp = get_config_value(interface, 'rsn_pairwise')
+        self.assertEqual('CCMP GCMP', tmp)
+        tmp = get_config_value(interface, 'wpa_key_mgmt')
+        self.assertEqual('SAE', tmp)
+
+        # beamforming
+        tmp = get_config_value(interface, 'he_mu_beamformer')
+        self.assertEqual('1', tmp)
+        tmp = get_config_value(interface, 'he_su_beamformee')
+        self.assertEqual('0', tmp)
+        tmp = get_config_value(interface, 'he_mu_beamformer')
+        self.assertEqual('1', tmp)
+
+        # Check for running process
+        self.assertTrue(process_named_running('hostapd'))
+
     def test_wireless_hostapd_wpa_config(self):
         # Only set the hostapd (access-point) options
-        interface = 'wlan0'
+        interface = 'wlan1'
         phy = 'phy0'
         ssid = 'ssid'
         channel = '1'
         wpa_key = 'VyOSVyOSVyOS'
         mode = 'n'
         country = 'de'
 
+        # SSID, country-code and channel are already configured in self.setUpClass()
+        # Therefore, we must delete those here to check if commit will fail without it.
+        self.cli_delete(self._base_path + [interface, 'ssid'])
+        self.cli_delete(self._base_path + [interface, 'country-code'])
         self.cli_set(self._base_path + [interface, 'physical-device', phy])
         self.cli_set(self._base_path + [interface, 'type', 'access-point'])
         self.cli_set(self._base_path + [interface, 'mode', mode])
+        self.cli_set(self._base_path + [interface, 'channel', channel])
 
         # SSID must be set
         with self.assertRaises(ConfigSessionError):
             self.cli_commit()
         self.cli_set(self._base_path + [interface, 'ssid', ssid])
 
-        # Channel must be set
-        with self.assertRaises(ConfigSessionError):
-            self.cli_commit()
-        self.cli_set(self._base_path + [interface, 'channel', channel])
-
```

Comment:
Why was this removed?
This should trigger a raise ConfigError if the channel is not set. After we get the expected error, we set the channel. It was in the original smoketest
```
        # Channel must be set
        with self.assertRaises(ConfigSessionError):
            self.cli_commit()
        self.cli_set(self._base_path + [interface, 'channel', channel])
```

Reply:
There is a default value set by the XML in [line 571](https://github.com/vyos/vyos-1x/blob/4ff64d4a73e84cacad1db3029854d846f94b1b6e/interface-definitions/interfaces_wireless.xml.in#L571). 

Because of this, a `channel` directive would always be present in the config dictionary. This would then prevent the assertion to succeed, because no `ConfigSessionError` would be raised, as there _is_ a channel configured per default. I investigated this and found that the default setting had been present for longer (actually, I didn't find the exact commit) and I even guess it was already present in pre-XML times. Removing this default would have created a breaking change in the CLI against existing configurations. I therefore opted against removing the default value, but _fixed the test_ instead which was expecting an exception raised because no channel would be configured - which can never happen with a default value. 

In fact, I am sure that it was exactly _this_ assert in the lines you marked, not the assert checking for SSID, which made the test fail before. I pinned it down by commenting each of the three asserts, one at a time, re-run the test and see if the test still failed.